code/modules/mob/living/basic/icemoon/ice_whelp/ice_whelp_ai.dm

```diff
@@ -0,0 +1,217 @@
+#define BB_TARGET_TREE "BB_target_tree"
+
+/datum/ai_controller/basic_controller/ice_whelp
+	blackboard = list(
+		BB_TARGETTING_DATUM = new /datum/targetting_datum/basic/allow_items/goliath,
+		BB_WHELP_ENRAGED = 0,
+	)
+
+	ai_movement = /datum/ai_movement/basic_avoidance
+	idle_behavior = /datum/idle_behavior/idle_random_walk
+	planning_subtrees = list(
+		/datum/ai_planning_subtree/simple_find_target,
+		/datum/ai_planning_subtree/targeted_mob_ability/ice_whelp,
+		/datum/ai_planning_subtree/attack_obstacle_in_path,
+		/datum/ai_planning_subtree/basic_melee_attack_subtree,
+		/datum/ai_planning_subtree/sculpt_statues,
+		/datum/ai_planning_subtree/cannibalize,
+		/datum/ai_planning_subtree/burn_trees,
+	)
+
+///subtree to find corpses of our kind and consume them
+/datum/ai_planning_subtree/cannibalize
```

Comment:
I feel like this re-implements what `find_and_hunt_target` already implements

That is to say you might be able to just use that tree. Set `targets` as our ai controller pawn type, set hunting behavior = `/datum/ai_behavior/cannibalize`.